parser/parser_test.go

```diff
@@ -7142,3 +7142,21 @@ func TestTTLTableOption(t *testing.T) {
 
 	RunTest(t, table, false)
 }
+
+func TestMultiStmt(t *testing.T) {
+	p := parser.New()
+	stmts, _, err := p.Parse("SELECT 'foo'; SELECT 'bar','baz' ; select 'foo' , 'bar', 'baz';select 1", "", "")
```

Comment:
Maybe a fieldname that contains a `;` as well?

```
mysql> SELECT 'foo;bar';
+---------+
| foo;bar |
+---------+
| foo;bar |
+---------+
1 row in set (0.00 sec)
```

Reply:
Done :)